packages/xstate-react/src/useSelector.ts

```diff
@@ -9,12 +9,17 @@ function isService(actor: any): actor is Interpreter<any, any, any, any> {
 }
 
 const defaultCompare = (a, b) => a === b;
-const defaultGetSnapshot = (a) =>
-  isService(a)
-    ? getServiceSnapshot(a)
-    : isActorWithState(a)
-    ? a.state
-    : undefined;
+const defaultGetSnapshot = (a, initialStateCacheRef) => {
+  if (isService(a)) {
+    if (a.status === 0 && initialStateCacheRef.current) {
+      return initialStateCacheRef.current;
+    }
+    const snapshot = getServiceSnapshot(a);
+    initialStateCacheRef.current = a.status === 0 ? snapshot : null;
+    return snapshot;
+  }
+  return isActorWithState(a) ? a.state : undefined;
+};
```

Comment:
Is it pretty? Not at all. It's basically a hack to cache those results as they should represent the same thing, even if they are not preferentially transparent. 

We can't easily refactor this in v4 because we can't compute the initial "true" snapshot when the service isn't initialized as computing this might lead to side-effects. This problem doesn't exist in v5